pkg/node/store/store.go

```diff
@@ -81,16 +87,27 @@ func (o *NodeObserver) OnDelete(k store.NamedKey) {
 	if n, ok := k.(*node.Node); ok {
 		nodeCopy := n.DeepCopy()
 		nodeCopy.Source = node.FromKVStore
-		o.manager.NodeDeleted(*nodeCopy)
 
-		ciliumIPv4 := nodeCopy.GetCiliumInternalIP(false)
-		if ciliumIPv4 != nil {
-			ipcache.IPIdentityCache.Delete(ciliumIPv4.String(), ipcache.FromKVStore)
-		}
-		ciliumIPv6 := nodeCopy.GetCiliumInternalIP(true)
-		if ciliumIPv6 != nil {
-			ipcache.IPIdentityCache.Delete(ciliumIPv6.String(), ipcache.FromKVStore)
-		}
+		func() {
+			time.Sleep(defaults.NodeDeleteDelay)
+
+			if o.manager.Exists(nodeCopy.Identity()) {
+				log.Warningf("Received node delete event for node %s which re-appeared within %s",
+					nodeCopy.Name, defaults.NodeDeleteDelay)
+				return
+			}
+
+			o.manager.NodeDeleted(*nodeCopy)
```

Comment:
Is there a log in this case? It can be really useful to have a log with the actual timestamp of this event when trying to debug any connectivity issues related to node deletion.

Reply:
In the next function:
```
func (m *Manager) NodeDeleted(n node.Node) {
        m.metricEventsReceived.WithLabelValues("delete", string(n.Source)).Inc()

        log.Debugf("Received node delete event from %s", n.Source)
```

Reply:
Maybe we should change the log level to `info`?

Reply:
It results in a ton of noise in large clusters, we changed a lot of this to debug as part of the scale work.

Reply:
In general, we avoid Info for state changes outside of the local node.